tests/php/sync/test_class.jetpack-sync-themes.php

```diff
@@ -123,45 +123,56 @@ public function test_changed_theme_is_synced() {
 		$this->assertEquals( $local_value, $this->server_replica_storage->get_option( 'theme_mods_' . $this->theme ) );
 	}
 
-	public function test_theme_install_and_delete() {
-		require_once ABSPATH . 'wp-admin/includes/theme-install.php';
-		require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
-
-		$theme_stylesheet = 'itek';
+	public function test_install_edit_delete_theme_sync() {
+		$theme_slug = 'itek';
 		$theme_name = 'iTek';
 
-		$api = themes_api(
-			'theme_information',
-			array(
-				'slug'   => $theme_stylesheet,
-			)
-		);
+		delete_theme( $theme_slug ); //Ensure theme is not lingering on file system
+		$this->server_event_storage->reset();
 
-		if ( is_wp_error( $api ) ) {
-			wp_die( $api );
-		}
-
-		$upgrader = new Theme_Upgrader( new Test_Upgrader_Skin( compact('title', 'nonce', 'url', 'theme') ) );
-		$upgrader->install( $api->download_link );
+		//Test Install Theme
 
+		$this->install_theme( $theme_slug );
 		$this->sender->do_sync();
+
 		$event_data = $this->server_event_storage->get_most_recent_event( 'jetpack_installed_theme' );
 
-		$this->assertEquals( $event_data->args[0], $theme_stylesheet );
+		$this->assertEquals( $event_data->args[0], $theme_slug );
 		$this->assertEquals( $event_data->args[1]['name'], $theme_name );
 		$this->assertTrue( (bool) $event_data->args[1]['version'] );
 		$this->assertTrue( (bool) $event_data->args[1]['uri'] );
 
-		delete_theme( $theme_stylesheet );
+		//Test Edit Theme
 
+		/**
+		 * This filter is already documented in wp-includes/pluggable.php
+		 *
+		 * @since 1.5.1
+		 */
+		$_POST['newcontent'] = 'foo';
+		apply_filters( 'wp_redirect', 'theme-editor.php?file=style.css&theme=' . $theme_slug . '&scrollto=0&updated=true');
```

Comment:
Should we try testing this with different types of urls passed into the redirect? 
Such as the full url?